hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/file/OMFileRequest.java

```diff
@@ -932,4 +934,50 @@ public static long getParentID(long bucketId, Iterator<Path> pathComponents,
 
     return lastKnownParentId;
   }
+
+  /**
+   * get a list of keys for checking acls.
+   * Returns a list only if the key is a directory and has children.
+   * Otherwise returns an empty list
+   * @param ozoneManager
+   * @param volumeName
+   * @param bucketName
+   * @param keyName
+   * @return
+   * @throws IOException
+   */
+  public static List<String> getKeyListForAclCheck(OzoneManager ozoneManager,
```

Comment:
Can we provide a API support batching as the the number of keys under recursive check can be huge?